hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/security/RootCARotationManager.java

```diff
@@ -219,6 +219,17 @@ public void start() throws SCMServiceException {
     LOG.info("Monitor task for root certificate {} is started with " +
         "interval {}.", scmCertClient.getCACertificate().getSerialNumber(),
         checkInterval);
+    executorService.scheduleAtFixedRate(this::removeExpiredCertTask, 0,
+        secConf.getExpiredCertificateRemovalInterval().toMillis(),
+        TimeUnit.MILLISECONDS);
+    LOG.info("Scheduling expired certificate removal with interval {}s",
+        secConf.getExpiredCertificateRemovalInterval().getSeconds());
+  }
+
+  private void removeExpiredCertTask() {
+    if (scm.getCertificateStore() != null) {
```

Comment:
Please add the isRunning check. 
if (!isRunning.get()) {
        return;
      }